### DISCUSSION TITLE
Merge Queue Feedback: Allow editing the commit message when adding to merge queue when "Squash & merge" is enabled

### DISCUSSION THREAD

**Discussion by wshanks** [community/community#111224] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

It would be ideal if the GitHub UI presented an editor view for editing the commit message of a PR when it is added to the merge queue when "Squash & merge" is selected. When not using the merge queue, selecting "Squash and merge" on a PR presents such an editor. I would like to see the same thing for the merge queue.

This request has been mentioned in a couple other discussions but I just wanted to call it out as its own discussion for greater visibility. The current behavior is that the merge queue uses the default commit message. As a workaround, I set the default commit message to be "Pull request title and description" and require the pull request author to keep the pull request description accurate and appropriate for a final commit message, but this flow is not ideal. Not doing this and keeping the default setting for the default commit message results in text from all the commit messages getting concatenated together in the merge queue commit's commit message. Since the GitHub UI encourages pushing fix up commits in response to code review rather than force pushing over the original commit, pull request branches tend to have many small ephemeral commits with throwaway messages and these add noise to the final commit message with the default settings.

**Comment by scottsand-db** [comment]
+1
